src/openforms/validations/registry.py

```diff
@@ -58,7 +40,7 @@ def _flat(it):
     return list(_flat(iterables))
 
 
-class Registry(BaseRegistry[RegisteredValidator]):
+class Registry(BaseRegistry[BasePlugin[T]]):
```

Comment:
Not sure having this nested type variable makes sense. I'm currently using it a few lines below:

```python
    def validate(
        self, plugin_id: str, value: T, submission: Submission
    ) -> ValidationResult:
```
But it will never be bound anyway. I guess it makes more sense to switch to:

```suggestion
class Registry(BaseRegistry[BasePlugin]):
```

Reply:
yeah we typically don't instantiate `Registry` at run-time, it's just a sentinel where any type can be provided, so I don't see added value in providing the typevar.

Plugin is now a base class, but effectively it's more of a protocol with specializations for possible value types.